PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrFragment.java

```diff
@@ -523,7 +523,15 @@ protected void onPostExecute(String[] result) {
 
                 }
             } else {
```

Comment:
Have you tried removing the try-catch and simply writing

Toast.makeText(getContext(), "Failed to retrieve GSR data", Toast.LENGTH_LONG).show();